File: src/vct/col/ast/AssignmentStatement.java
// -*- tab-width:2 ; indent-tabs-mode:nil -*-
package vct.col.ast;

import hre.ast.MessageOrigin;

import java.util.*;


public class AssignmentStatement extends ASTNode {

  private ASTNode expression;
  private ASTNode location;

  public AssignmentStatement(String name,ASTNode expression){
    location=new NameExpression(name);
    location.setOrigin(new MessageOrigin("parser bug: string location in assignment"));
    this.expression=expression;
  }

  public AssignmentStatement(ASTNode location,ASTNode expression){
    this.location=location;
    this.expression=expression;
  }

  public ASTNode getExpression() { return expression; }

//  public String getName() { return name; }

  public ASTNode getLocation() { return location; }

  public void accept_simple(ASTVisitor visitor){
    visitor.visit(this);
  }

}

File: vercors/src/main/java/vct/col/ast/stmt/terminal/AssignmentStatement.scala
package vct.col.ast.stmt.terminal

import hre.ast.MessageOrigin
import vct.col.ast._
import vct.col.ast.expr.NameExpression
import vct.col.ast.generic.ASTNode
import vct.col.ast.util.ASTVisitor
import vct.col.util.{ASTMapping, ASTMapping1}

case class AssignmentStatement(val location:ASTNode, val expression:ASTNode) extends ASTNode {

  def this(name:String, expression:ASTNode) = {
    this(new NameExpression(name), expression)
    location.setOrigin(new MessageOrigin("parser bug: string location in assignment"))
  }

  override def accept_simple[T,A](m:ASTMapping1[T,A], arg:A) = m.map(this, arg)
  override def accept_simple[T](v:ASTVisitor[T]) = v.visit(this)
  override def accept_simple[T](m:ASTMapping[T]) = m.map(this)

  override def debugTreeChildrenFields(): Iterable[String] = Seq("location", "expression")
  override def debugTreePropertyFields(): Iterable[String] = Seq()
}
